scripts/lldb/.lldb/o3destd_lldb.py

```diff
@@ -0,0 +1,232 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+# See the following pages for information on how to write formatters for lldb
+# https://lldb.llvm.org/use/variable.html#finding-formatters-101
+# https://lldb.llvm.org/python_api/lldb.SBValue.html#lldb.SBValue
+# https://github.com/llvm/llvm-project/blob/main/lldb/examples/synthetic/libcxx.py
+import enum
+import lldb
+import lldb.formatters.Logger
+
+# Uncomment for debug output
+# lldb.formatters.Logger._lldb_formatters_debug_level = 3
+
+def read_string_data(process: lldb.SBProcess, string_type: lldb.SBType, string_address: lldb.SBValue, string_size: int):
+    # If the string size is 0, then return an empty string
+    if string_size == 0:
+        return '""'
+
+    # Get the string element type from the first tempalte argument
+    element_type = string_type.GetTemplateArgumentType(0)
+    # query the byte size of the character type. It should be 1 for `char`
+    # 2 for wchar_t on Windows and 4 for wchar_t on Linux/MacOS
+    element_size = element_type.GetByteSize()
+
+    # Stores the encoding to use to decode the string byte data read in from memory
+    # into a python unicode string
+    encoding = ''
+
+    # SBType.GetBasicType is strange, passing in the enum value of the type
+    # doesn't query the actual type but returns a new SBType
+    # instance that represents the basic type
+    # https://lldb.llvm.org/python_api/lldb.SBType.html#lldb.SBType.GetBasicType
+    if element_type == element_type.GetBasicType(lldb.eBasicTypeChar):
+        encoding = 'utf-8'
+    elif element_type == element_type.GetBasicType(lldb.eBasicTypeWChar):
+        if element_size == 4:
+            byte_data = process.ReadMemory(string_address.GetValueAsUnsigned(), string_size * element_size, error)
```

Comment:
does `byte_data` need to be read here? (and line 46). Because it is eventually read below on line 58. (unless its used to move the stream pointer forward or something)

Reply:
No, these calls should be removed.